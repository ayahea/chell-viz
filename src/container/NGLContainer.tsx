import * as NGL from 'ngl';
import * as React from 'react';
import * as util from 'util';

import { IStageLoadFileParams, PickingProxy, Stage, StructureComponent } from 'ngl';
import { GeneInfo } from '../component/GeneInfo';

export class NGLContainer extends React.Component<any, any> {
  private canvas: HTMLElement | null = null;
  constructor(props: any) {
    super(props);
  }

  public async componentDidMount() {
    if (this.canvas) {
      const stage = new NGL.Stage(this.canvas);
      const params: IStageLoadFileParams = {
        defaultRepresentation: true,
      };

      const structure = (await stage.loadFile('assets/1fqg.pdb', params)) as StructureComponent;

      structure.autoView();

      let ele: NGL.RepresentationElement;

      stage.mouseControls.add('hoverPick', (aStage: Stage, pickingProxy: PickingProxy) => {
        if (pickingProxy && (pickingProxy.atom || pickingProxy.bond)) {
          const atom = pickingProxy.atom || pickingProxy.closestBondAtom;
          if (ele) {
            structure.removeRepresentation(ele);
          }
          ele = structure.addRepresentation('spacefill', {
            sele: atom.resno.toString(),
          });
        }
      });
    }
  }

  public render() {
    return (
      <div id="NGLContainer">
        <div ref={el => (this.canvas = el)} style={{ height: 400, width: 400 }} />
        <GeneInfo />
      </div>
    );
  }
}
