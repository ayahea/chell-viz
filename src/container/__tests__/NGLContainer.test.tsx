import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { NGLContainer } from '../NGLContainer';

describe('NGLContainer', () => {
  test('Should match existing snapshot when given no props.', () => {
    expect(renderer.create(<NGLContainer />)).toMatchSnapshot();
  });
});
