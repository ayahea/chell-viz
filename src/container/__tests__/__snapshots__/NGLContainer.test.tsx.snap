// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NGLContainer Should match existing snapshot when given no props. 1`] = `
<div
  id="NGLContainer"
>
  <div
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
  />
</div>
`;
